agent/agent.go

```diff
@@ -684,6 +695,27 @@ func (a *Agent) Start(ctx context.Context) error {
 		{Name: "pre_release", Value: a.config.VersionPrerelease},
 	})
 
+	// start a go routine to reload config based on file watcher events
+	if a.baseDeps.RuntimeConfig.AutoReloadConfig {
+		if a.baseDeps.WatchedFiles != nil {
+			w, err := config.NewFileWatcher(a.baseDeps.WatchedFiles, a.baseDeps.Logger)
+			if err != nil {
+				a.baseDeps.Logger.Error("error loading config", "error", err)
+			}
+			a.FileWatcher = w
+			go func() {
+				for event := range a.FileWatcher.EventsCh {
+					a.baseDeps.Logger.Debug("auto-reload config triggered", "event-file", event.Filename)
+					err := a.ReloadConfig(true)
+					if err != nil {
+						a.baseDeps.Logger.Error("error loading config", "error", err)
+					}
+				}
+			}()
+			a.baseDeps.Logger.Debug("starting file watcher")
+			a.FileWatcher.Start(context.Background())
```

Comment:
Seems a little strange to start _after_ running the goroutine to consume the events that aren't being generated yet.